src/button-dropdown/__tests__/button-dropdown.test.tsx

```diff
@@ -256,19 +256,35 @@ describe('with main action', () => {
     expect(onFollow).toHaveBeenCalledTimes(1);
   });
 
-  test('main action with external link has an icon and dedicated ARIA label', () => {
+  test('main action assigns ARIA label', () => {
     const wrapper = renderSplitButtonDropdown({
       mainAction: {
         text: 'Main',
-        href: 'https://external.com',
-        external: true,
-        externalIconAriaLabel: '(opens in a new tab)',
+        ariaLabel: 'Main #1',
       },
     });
 
+    expect(wrapper.findMainAction()!.getElement()).toHaveTextContent('Main');
+    expect(wrapper.findMainAction()!.getElement()).toHaveAccessibleName('Main #1');
+  });
+
+  test('main action with external link has an icon and dedicated ARIA label', () => {
+    const mainAction = {
+      text: 'Main',
+      href: 'https://external.com',
+      external: true,
+      externalIconAriaLabel: '(opens in a new tab)',
+    };
+    const wrapper = renderSplitButtonDropdown({ mainAction });
+
     expect(wrapper.findMainAction()?.findByClassName(iconStyles.icon)).not.toBe(null);
     expect(wrapper.findMainAction()!.getElement()).toHaveTextContent('Main');
     expect(wrapper.findMainAction()!.getElement()).toHaveAccessibleName('Main (opens in a new tab)');
+
+    const wrapper2 = renderSplitButtonDropdown({ mainAction: { ...mainAction, ariaLabel: 'Main #1' } });
```

Comment:
Is there any reason that wrapper2 is not its own test case? 

I mean something like this:
```tsx
describe('when aria label is provided', () => {
  it('assigns the provided aria label', () => { ... });
}

describe('when aria label is not provided', () => {
  it('assigns the default aria label', () => { ... });
}
```

Reply:
I don't have a specific preference - updated to have two test cases